measure_alerts.sh

```diff
@@ -0,0 +1,59 @@
+SLEEP_INTERVAL_SECONDS=1
+
+function die() {
+	local m="${1}"
+	echo "[$(date)] FATAL: ${m}" >&2
+	exit 1
+}
+
+function message {
+    local m="${1}"
+    echo "[$(date)] INFO: ${m}"
+}
+
+function output_time() {
+    local seconds_start="${1}"
+    local seconds_now=$(date +%s)
+    local now=$(date -d @${seconds_now} +"%T")
+    message "$((seconds_now-seconds_start)) Seconds Passed"
+}
+
+#
+# detect_changes(topic, sampling expression)
+#
+function detect_changes() {
+    local topic="${1}"
+    local expression="${2}"
+    message "Waiting For change: TOPIC=${topic}"
+    message "Waiting For change: EXPRESSION=${expression}"
+    local initial_value=$(eval "${expression}")
+    local current_value="${initial_value}"
+    message "Initial Value is ${initial_value}"
+    message "Waiting..."
+    while true; do
+        current_value=$(eval "${expression}")
+        if [[ ${current_value} != ${initial_value} ]]; then
```

Comment:
shellcheck complains about this line:

```
        if [[ ${current_value} != ${initial_value} ]]; then
                                  ^-- SC2053: Quote the rhs of != in [[ ]] to prevent glob matching.

```
See: https://github.com/koalaman/shellcheck/wiki/SC2053

I think you can quote one of the vars like that wikipage suggests, or use
` if [ "${current_value} != ${initial_value}" ]; then `
(single brackets instead of double brackets).

Also (sorry for nitpicking), is there any reason you're doing it in an `if` instead of putting the condition in the `while`? I think that way you could avoid the `break` statement and it'd be a bit more readable.